### ISSUE TITLE
Route costing for residential roads

### ISSUE THREAD

**Issue by ArterialMapping** [valhalla/valhalla#3625]
Is there a way to add costing to a route request so that the route avoids certain types of roadways such as residential? 

I know I can avoid Highway, but did not see any documentation for roadways.

**Comment by nilsnolde** [comment]
no sorry, we don't have a generic way to avoid road types from routing over them. we do have that for snapping the waypoints (origin/via/destination), that won't help you much though I think.

**Comment by ArterialMapping** [comment]
> no sorry, we don't have a generic way to avoid road types from routing over them. we do have that for snapping the waypoints (origin/via/destination), that won't help you much though I think.

Got it, this would be cool functionality for the future. But I should be able to generate avoid polygons as a workaround for now (I think).

**Comment by kevinkreiser** [comment]
yeah there is a lot of stuff that we dont allow complete control over in the costing. honestly i would love it if we just opened all of the parameters up in costing, had a single costing class with some presets to select certain things like mountain bike or truck etc. the thing is that would be a major revision but at least it would make it possible to never have to answer a question with the answer we have to give here which is: 

> well, we hardcode per-road class per-costing an affinity for certain types of roads and you cant touch it from the user-facing api.
